src/charm.py

```diff
@@ -186,22 +190,47 @@ def _on_install(self, _) -> None:
         except (apt.InvalidSourceError, ValueError, apt.GPGKeyError, URLError):
             self.unit.status = BlockedStatus("couldn't install MongoDB")
 
+        # Construct the mongod startup commandline args for systemd
+        machine_ip = self._unit_ip(self.unit)
+        mongod_start_args = " ".join(
+            [
+                "ExecStart=/usr/bin/mongod",
+                # bind to localhost and external interfaces
+                "--bind_ip",
+                f"localhost,{machine_ip}",
+                # part of replicaset
+                "--replSet",
+                "rs0",
+                "\n",
+            ]
+        )
+
+        with open("/lib/systemd/system/mongod.service", "r") as service_template:
+            mongodb_service = service_template.readlines()
+
+        # replace start command with our parameterized one
+        for i in range(len(mongodb_service)):
+            if "ExecStart" in mongodb_service[i]:
+                mongodb_service[i] = mongod_start_args
+
+        # systemd gives files in /etc/systemd/system/ precedence over those in /lib/systemd/system/
+        # hence our changed file in /etc will be read while maintaining the original one in /lib.
+        with open("/etc/systemd/system/mongod.service", "w") as service_file:
+            service_file.writelines(mongodb_service)
+
+        # mongod requires permissions to /data/db
+        mongodb_user = pwd.getpwnam(MONGO_USER)
+        os.chown(MONGO_DATA_DIR, mongodb_user.pw_uid, mongodb_user.pw_gid)
+
+        # changes to service files are only applied after reloading
+        systemd.daemon_reload()
+
```

Comment:
do we need to handle errors here?
like writing errors, chown, reload errors

Reply:
resolving due to our Mattermost chat.